src/app/tests/suites/certification/Test_TC_DM_2_1.yaml

```diff
@@ -13,91 +13,180 @@
+      command: "WaitForCommissionee"
+      arguments:
+          values:
+              - name: "nodeId"
+                value: nodeId
+
+    - label: "Query Data Model Revision"
+      command: "readAttribute"
+      attribute: "DataModelRevision"
+      response:
+          constraints:
+              type: uint16
+
+    - label: "Query Vendor Name"
+      command: "readAttribute"
+      attribute: "VendorName"
+      response:
+          constraints:
+              type: string
+              maxLength: 32
+
+    - label: "Query VendorID"
+      command: "readAttribute"
+      attribute: "VendorID"
+      response:
+          constraints:
+              type: uint16
+
+    - label: "Query Product Name"
+      command: "readAttribute"
+      attribute: "ProductName"
+      response:
+          constraints:
+              type: string
+              maxLength: 32
+
+    - label: "Query ProductID"
+      command: "readAttribute"
+      attribute: "ProductID"
+      response:
+          constraints:
+              type: uint16
+
+    - label: "Query Node Label"
+      command: "readAttribute"
+      attribute: "NodeLabel"
+      response:
+          constraints:
+              type: string
+              maxLength: 32
+
+    - label: "Query User Location"
+      command: "readAttribute"
+      attribute: "Location"
+      response:
+          constraints:
+              type: string
+              maxLength: 2
+              format: "ISO 3166-1 alpha-2" # Step 6 11.2.3.1-6 Verify it follows ISO 3166-1 alpha-2 code / or it is empty, since it is user-defined
+
+    - label: "Query HardwareVersion"
+      command: "readAttribute"
+      attribute: "HardwareVersion"
+      response:
+          constraints:
+              type: uint16
+
+    - label: "Query HardwareVersionString"
+      command: "readAttribute"
+      attribute: "HardwareVersionString"
+      response:
+          constraints:
+              type: string
+              minLength: 1
+              maxLength: 64
+
+    - label: "Query SoftwareVersion"
+      command: "readAttribute"
+      attribute: "SoftwareVersion"
+      response:
+          constraints:
+              type: uint32
+
+    - label: "Query SoftwareVersionString"
+      command: "readAttribute"
+      attribute: "SoftwareVersionString"
+      response:
+          constraints:
+              type: string
+              minLength: 1
+              maxLength: 64
+              format: "ASCII" # Step 10 11.2.3.1-10 Verify it uses 7-bit ASCII alphanumeric and punctuation characters
+
+    - label: "Query ManufacturingDate"
+      command: "readAttribute"
+      attribute: "ManufacturingDate"
+      PICS: MANF_DATE
+      optional: true
+      response:
+          constraints:
+              type: string
+              minLength: 8
+              maxLength: 16
+              format: "ISO 8601" # Step 11  11.2.3.1-11 Verify if the first 8 characters specify date according to ISO 8601, i.e, YYYYMMDD
```

Comment:
It will not be ISO8601, only the first 8 charactres will be.. We should have a "format_regex" to allow us to write regex for validations

Reply:
currently yaml framework is not supporting this format checking ,hence i changed this particular step as User Prompt for time being